zipline/algorithm.py

```diff
@@ -705,6 +579,43 @@ def get_environment(self, field='platform'):
         else:
             return env[field]
 
+    @api_method
+    def fetch_csv(self, url,
+                  pre_func=None,
+                  post_func=None,
+                  date_column='date',
+                  date_format=None,
+                  timezone=pytz.utc.zone,
+                  symbol=None,
+                  mask=True,
+                  symbol_column=None,
+                  special_params_checker=None,
+                  **kwargs):
+
+        # Show all the logs every time fetcher is used.
+        csv_data_source = PandasRequestsCSV(
+            url,
+            pre_func,
+            post_func,
+            self.trading_environment,
+            self.sim_params.period_start,
+            self.sim_params.period_end,
+            date_column,
+            date_format,
+            timezone,
+            symbol,
+            mask,
+            symbol_column,
+            data_frequency=self.data_frequency,
+            special_params_checker=special_params_checker,
+            **kwargs
+        )
+
+        # ingest this into dataportal
+        self.data_portal.handle_extra_source(csv_data_source.df)
```

Comment:
How does this update to the data portal state intersect with multiple calls to `run`?  Will the extra source persist?